docs/content/operations/tls-support.md

```diff
@@ -0,0 +1,50 @@
+---
+layout: doc_page
+---
+
+TLS Support
+===============
+
+# General Configuration
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.server.http.plainText`|Enable/Disable HTTP connector.|`true`|
+|`druid.server.http.tls`|Enable/Disable HTTPS connector.|`false`|
+
+Although not recommended but both HTTP and HTTPS connectors can be enabled at a time and respective ports are configurable using `druid.plainTextPort`
+and `druid.tlsPort` properties on each node. Please see `Configuration` section of individual nodes to check the valid and default values for these ports.
+
+# Jetty Server TLS Configuration
+
+Druid uses Jetty as an embedded web server. To get familiar with TLS/SSL in general and related concepts like Certificates etc.
+reading this [Jetty documentation](http://www.eclipse.org/jetty/documentation/9.3.x/configuring-ssl.html) might be helpful.
+To get more in depth knowledge of TLS/SSL support in Java in general, please refer to this [guide](http://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html).
+The documentation [here](http://www.eclipse.org/jetty/documentation/9.3.x/configuring-ssl.html#configuring-sslcontextfactory)
+can help in understanding TLS/SSL configurations listed below. This [document](http://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html) lists all the possible
+values for the below mentioned configs among others provided by Java implementation.
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+|`druid.server.https.keyStorePath`|The file path or URL of the TLS/SSL Key store.|`null`|yes|
+|`druid.server.https.keyStoreType`|The type of the key store.|`null`|yes|
+|`druid.server.https.certAlias`|Alias of TLS/SSL certificate for the connector.|`null`|yes|
+
+All these configs must be set to some valid value. Apart from these properties, `keyStorePassword` and `keyManagerPassword` are required
```

Comment:
`PasswordProvider` has been the interface used for getting metadata store password. `DefaultPasswordProvider` is the implementation used for fetching this password from runtime properties file. Users have to implement their own `PasswordProvider` in an extension id they want to has their own way of fetching the password.
Currently, this interface has just one method `getPassword()` which just assumes that it will be used to fetch metadata store password.
For the TLS support, Jetty server needs to access the key store which is password protected. I wanted a way to fetch this password which is extensible so that users can implement their own way for fetching pwd. So, I extended the `PasswordProvider` to include `getPassword(String key)` where key would be the thing using which the implementation decides which password to fetch. So, `keyStorePassword ` and `keyManagerPassword` are the keys. 
Probably, the confusion is `PasswordProvider` interface is not documented enough ? Do you have a better design in mind to achieve this functionality ?